### PR TITLE
Addons pulled from API args

### PR DESCRIPTION
**Pull request by unclekingpin**
send diff of transport_urls in AddonsPulledFromAPI event args

### CODE REVIEW

File: src/models/ctx/update_profile.rs [hunk 1]
```diff
@@ -82,10 +83,14 @@ pub fn update_profile<E: Env + 'static>(
...
-                    .cloned()
+                let local_addons = profile.addons.iter().cloned().collect::<HashSet<_>>();
+                let remote_addons = next_addons.iter().cloned().collect::<HashSet<_>>();
+                let added_addons = &remote_addons - &local_addons;
+                let removed_addon = &local_addons - &remote_addons;
+                let transport_urls = added_addons
+                    .into_iter()
+                    .chain(removed_addon.into_iter())
+                    .map(|addon| addon.transport_url)
                     .collect();
```
Comment (elpiel) [inline src/models/ctx/update_profile.rs]: Should we try to deduplicate the result to remove duplicates?

**Review by elpiel — APPROVED** [review]
I only have 1 small comment for the deduplication and I would like to discuss it before merging.

File: src/models/ctx/update_profile.rs [hunk 2]
```diff
@@ -82,10 +83,22 @@ pub fn update_profile<E: Env + 'static>(
...
                     .iter()
                     .map(|addon| &addon.transport_url)
                     .cloned()
+                    .collect::<HashSet<_>>();
+                let added_transport_urls = &next_transport_urls - &prev_transport_urls;
+                let removed_transport_urls = &prev_transport_urls - &next_transport_urls;
+                let transport_urls = added_transport_urls
+                    .into_iter()
+                    .chain(removed_transport_urls.into_iter())
                     .collect();
```
Comment (elpiel) [inline src/models/ctx/update_profile.rs:102]: This could be moved to it's own function as it repeats with the lines below.

**Review by elpiel — APPROVED** [review]

File: src/models/ctx/update_profile.rs [hunk 3]
```diff
@@ -82,10 +83,22 @@ pub fn update_profile<E: Env + 'static>(
...
+                    .map(|addon| &addon.transport_url)
+                    .cloned()
+                    .collect::<HashSet<_>>();
+                let next_transport_urls = next_addons
                     .iter()
                     .map(|addon| &addon.transport_url)
                     .cloned()
+                    .collect::<HashSet<_>>();
+                let added_transport_urls = &next_transport_urls - &prev_transport_urls;
+                let removed_transport_urls = &prev_transport_urls - &next_transport_urls;
```
Comment (elpiel) [inline src/models/ctx/update_profile.rs:98]: Why do we need to add the removed `transport_url`s to the `AddonsPulledFromAPI` event?

Comment (unclekingpin) [reply src/models/ctx/update_profile.rs:98]: the idea is to receive the **changed** transport urls in the event arg. so we know that if the vec is not empty something was changed

**Pull request merged**